### PR TITLE
Typescript: Add .d.ts typedef

### PR DESCRIPTION
**Pull request by weikinhuang**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      href: string,
+    },
+  }
+
+  type Connection = string | {
+    name: string,
+    data: KeyValuePair,
+  };
+
+  export type HippoRequestOptions = RequestInit & KeyValuePair;
```
Comment (weikinhuang) [inline index.d.ts]: Fetch request options: https://github.com/Microsoft/TypeScript/blob/a6555921f4b1d51281b1241c69e91e5b82d2f07e/lib/lib.webworker.d.ts#L93

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
+declare module 'be-hippo' {
+
+  type MIME_TYPE_JSON = 'application/json';
```
Comment (erindepew) [inline index.d.ts]: if these are only being used in one place, is it necessary to define types?

Comment (weikinhuang) [reply index.d.ts]: that's fair, since i'm not currently exporting it

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  type KeyValuePair = { [k: string]: any };
+
+  type Descriptor = {
+    [name: string]: {
+      accept?: ACCEPTED_MIME_TYPE[],
+      href: string,
+    },
+  }
+
+    export type Connection = string | {
```
Comment (eerikson) [inline index.d.ts]: Think this is indented one too many.

Comment (weikinhuang) [reply index.d.ts]: fixed, thanks!

**Pull request merged**